### PR TITLE
tests/posix_sleep: fix for invalid RTT configurations 

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

While testing https://github.com/RIOT-OS/Release-Specs/issues/202 I found that `posix_sleep` was failing because of invalid RTT configurations. The proper fix is different than the one proposed here, I think it would mean adding a feature, but for a realease backport this is IMO the smallest change.

### Testing procedure

Test on `iotab-m3` now passes same for any `kinetis` or `efm32` `BOARD`.

### CODE REVIEW

File: tests/posix_sleep/Makefile [hunk 1]
```diff
@@ -2,8 +2,12 @@ include ../Makefile.tests_common
 
 USEMODULE += posix_sleep
 
-# Pull-in periph-rtt on board that provides this feature to switch to the RTT
-# backend of ztimer
-FEATURES_OPTIONAL += periph_rtt
+# This CPU families have a non configurable RTT of 1Hz, not enough for
```
Comment (aabadie) [inline tests/posix_sleep/Makefile:5]:
```suggestion
# These CPU families have a non configurable RTT of 1Hz, not enough for
```

File: tests/posix_sleep/Makefile [hunk 2]
```diff
@@ -2,8 +2,12 @@ include ../Makefile.tests_common
...
+# ztimer_msec to run on periph_rtt
+ifeq (,$(filter efm32 kinetis,$(CPU)))
+  # Pull-in periph-rtt on board that provides this feature to switch to the RTT
```
Comment (aabadie) [inline tests/posix_sleep/Makefile:8]:
```suggestion
  # Pull-in periph-rtt on boards that provide this feature to switch to the RTT
```

**Review by aabadie — COMMENTED** [review]
Initial comments :)

**Comment by fjmolinas** [comment]
ping @aabadie :)

**Review by aabadie — APPROVED** [review]
<details><summary>works on iotlab-m3</summary>

```
make BOARD=iotlab-m3 -C tests/posix_sleep flash test IOTLAB_NODE=auto-ssh
make: Entering directory '/work/riot/RIOT/tests/posix_sleep'
Building application "tests_posix_sleep" for "iotlab-m3" with MCU "stm32".

"make" -C /work/riot/RIOT/boards/iotlab-m3
"make" -C /work/riot/RIOT/boards/common/iotlab
"make" -C /work/riot/RIOT/core
"make" -C /work/riot/RIOT/cpu/stm32
"make" -C /work/riot/RIOT/cpu/cortexm_common
"make" -C /work/riot/RIOT/cpu/cortexm_common/periph
"make" -C /work/riot/RIOT/cpu/stm32/periph
"make" -C /work/riot/RIOT/cpu/stm32/stmclk
"make" -C /work/riot/RIOT/cpu/stm32/vectors
"make" -C /work/riot/RIOT/drivers
"make" -C /work/riot/RIOT/drivers/periph_common
"make" -C /work/riot/RIOT/sys
"make" -C /work/riot/RIOT/sys/auto_init
"make" -C /work/riot/RIOT/sys/frac
"make" -C /work/riot/RIOT/sys/isrpipe
"make" -C /work/riot/RIOT/sys/malloc_thread_safe
"make" -C /work/riot/RIOT/sys/newlib_syscalls_default
"make" -C /work/riot/RIOT/sys/pm_layered
"make" -C /work/riot/RIOT/sys/posix/sleep
"make" -C /work/riot/RIOT/sys/stdio_uart
"make" -C /work/riot/RIOT/sys/test_utils/interactive_sync
"make" -C /work/riot/RIOT/sys/tsrb
"make" -C /work/riot/RIOT/sys/ztimer
   text	   data	    bss	    dec	    hex	filename
  12288	    176	   2520	  14984	   3a88	/work/riot/RIOT/tests/posix_sleep/bin/iotlab-m3/tests_posix_sleep.elf
iotlab-node --jmespath='keys(@)[0]' --format='int'  --list saclay,m3,1 --flash /work/riot/RIOT/tests/posix_sleep/bin/iotlab-m3/tests_posix_sleep.bin | grep 0
0
r
ssh -t abadie@saclay.iot-lab.info 'socat - tcp:m3-1.saclay.iot-lab.info:20000' 
READY
s
START
main(): This is RIOT! (Version: 2021.04-devel-68-gbafca-review_posix_sleep)
5 x usleep(i++ * 500000)
wake up
wake up
wake up
wake up
wake up
5 x sleep(i++)
wake up
wake up
wake up
wake up
wake up
DONE
```

</details>

<details><summary>works on arduino-zero</summary>

```
make BOARD=arduino-zero -C tests/posix_sleep flash test IOTLAB_NODE=auto-ssh
make: Entering directory '/work/riot/RIOT/tests/posix_sleep'
Building application "tests_posix_sleep" for "arduino-zero" with MCU "samd21".

"make" -C /work/riot/RIOT/boards/arduino-zero
"make" -C /work/riot/RIOT/boards/common/arduino-zero
"make" -C /work/riot/RIOT/core
"make" -C /work/riot/RIOT/cpu/samd21
"make" -C /work/riot/RIOT/cpu/cortexm_common
"make" -C /work/riot/RIOT/cpu/cortexm_common/periph
"make" -C /work/riot/RIOT/cpu/sam0_common
"make" -C /work/riot/RIOT/cpu/sam0_common/periph
"make" -C /work/riot/RIOT/cpu/samd21/periph
"make" -C /work/riot/RIOT/cpu/samd21/vectors
"make" -C /work/riot/RIOT/drivers
"make" -C /work/riot/RIOT/drivers/periph_common
"make" -C /work/riot/RIOT/sys
"make" -C /work/riot/RIOT/sys/auto_init
"make" -C /work/riot/RIOT/sys/frac
"make" -C /work/riot/RIOT/sys/isrpipe
"make" -C /work/riot/RIOT/sys/malloc_thread_safe
"make" -C /work/riot/RIOT/sys/newlib_syscalls_default
"make" -C /work/riot/RIOT/sys/pm_layered
"make" -C /work/riot/RIOT/sys/posix/sleep
"make" -C /work/riot/RIOT/sys/stdio_uart
"make" -C /work/riot/RIOT/sys/test_utils/interactive_sync
"make" -C /work/riot/RIOT/sys/tsrb
"make" -C /work/riot/RIOT/sys/ztimer
   text	   data	    bss	    dec	    hex	filename
  13088	    176	   2516	  15780	   3da4	/work/riot/RIOT/tests/posix_sleep/bin/arduino-zero/tests_posix_sleep.elf
iotlab-node --jmespath='keys(@)[0]' --format='int'  --list saclay,arduino-zero,1 --flash /work/riot/RIOT/tests/posix_sleep/bin/arduino-zero/tests_posix_sleep.bin | grep 0
0
r
ssh -t abadie@saclay.iot-lab.info 'socat - tcp:arduino-zero-1.saclay.iot-lab.info:20000' 
READY
s
START
main(): This is RIOT! (Version: 2021.04-devel-68-gbafca-review_posix_sleep)
5 x usleep(i++ * 500000)
wake up
wake up
wake up
wake up
wake up
5 x sleep(i++)
wake up
wake up
wake up
wake up
wake up
DONE
```

</details>

I don't have a kinetis and efm32 for testing but I trust your testing.

ACK

**Comment by aabadie** [comment]
@fjmolinas this one still needs a backport !

**Comment by fjmolinas** [comment]
Backport provided in #15876

**Pull request merged**